instance/models/instance.py

```diff
@@ -170,44 +197,62 @@ def on_pre_save(sender, instance, **kwargs):
             self.base_domain = settings.INSTANCES_BASE_DOMAIN
 
-    @property
-    def log_text(self):
+    @staticmethod
+    def _sort_log_entries(server_logs, instance_logs):
         """
-        Combines the instance and server log outputs in chronological order
-        Currently only supports one non-terminated server at a time
-        Returned as a text string
+        Helper method to combine the instance and server log outputs in chronological order
         """
-        current_server = self.active_server_set.get()
-        # TODO: Filter out log entries for which the user doesn't have view rights
-        server_log_entry_set = current_server.log_entry_set.order_by('pk')\
-                                                           .iterator()
-        instance_log_entry_set = self.log_entry_set.order_by('pk')\
-                                                   .iterator()
-
-        next_server_log_entry = partial(next, server_log_entry_set, None)
-        next_instance_log_entry = partial(next, instance_log_entry_set, None)
+        next_server_log_entry = partial(next, server_logs, None)
+        next_instance_log_entry = partial(next, instance_logs, None)
 
-        log_text = ''
+        log = []
         instance_log_entry = next_instance_log_entry()
         server_log_entry = next_server_log_entry()
 
         while instance_log_entry is not None and server_log_entry is not None:
             if server_log_entry.created < instance_log_entry.created:
-                log_text += '{}\n'.format(server_log_entry)
+                log.append(server_log_entry)
                 server_log_entry = next_server_log_entry()
             else:
-                log_text += '{}\n'.format(instance_log_entry)
+                log.append(instance_log_entry)
                 instance_log_entry = next_instance_log_entry()
 
         while instance_log_entry is not None:
-            log_text += '{}\n'.format(instance_log_entry)
+            log.append(instance_log_entry)
             instance_log_entry = next_instance_log_entry()
 
         while server_log_entry is not None:
-            log_text += '{}\n'.format(server_log_entry)
+            log.append(server_log_entry)
             server_log_entry = next_server_log_entry()
 
-        return log_text
+        return log
+
+    @property
+    def log_entries(self):
+        """
+        Return the list of log entry instances for the instance and its current active server
+        """
+        # TODO: Filter out log entries for which the user doesn't have view rights
+        server_log_entry_set = self._current_server.log_entry_set.order_by('pk')\
+                                                                 .iterator()
+        instance_log_entry_set = self.log_entry_set.order_by('pk')\
+                                                   .iterator()
+        return Instance._sort_log_entries(server_log_entry_set, instance_log_entry_set)
+
+    @property
+    def errors(self):
```

Comment:
@chromano To be consistent with the method name above, can this be `log_error_entries()`?

Actually they both have very similar code - could they be factored into a single method:

```
def get_log_entries(self, level_list=None):
```

which the properties would call?